### PR TITLE
Update release notes for version 1.5.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
### Description
Reorganize legacy release notes and add new one for the upcoming release.
 
### Issues Resolved
Update release notes for the upcoming release.
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by matthewryanwells — APPROVED** [review]

**Review by GumpacG — APPROVED** [review]

**Review by acarbonetto — APPROVED** [review]

File: release-notes/sql-odbc.OpenSearch.release-notes-1.5.0.0.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+## Version 1.5
```
[the remark this answers is not in the document]
Comment (Yury-Fridlyand) [reply release-notes/sql-odbc.OpenSearch.release-notes-1.5.0.0.md:1]: Yes. Maybe I'll place here a date of publishing release on GH.

File: release-notes/sql-odbc.OpenSearch.release-notes-1.5.0.0.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+### Features
+
+* Add new parameter for SIGv4 auth. ([#42](https://github.com/opensearch-project/sql-odbc/pull/42))
```
Comment (MaxKsyunz) [inline release-notes/sql-odbc.OpenSearch.release-notes-1.5.0.0.md:5]: Can you call out the parameter name so users don't have to read the docs?

**Review by MaxKsyunz — APPROVED** [review]

File: release-notes/sql-odbc.OpenSearch.release-notes-1.5.0.0.md (hunk 2 above)
Comment (Yury-Fridlyand) [reply release-notes/sql-odbc.OpenSearch.release-notes-1.5.0.0.md]: Fixed in f9f9042.

File: release-notes/sql-odbc.OpenSearch.release-notes-1.5.0.0.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+### Documentation
+
+* Add how to setup aws credentials for ODBC Tableau ([#394](https://github.com/opensearch-project/sql/pull/394))
+* Update BI connectors and drivers readme files ([#665](https://github.com/opensearch-project/sql/pull/665))
+
+### Maintenance
+
+* Revert to windows 2019 for odbc CI ([#413](https://github.com/opensearch-project/sql/pull/413))
+* Version bump to 1.3 ([#419](https://github.com/opensearch-project/sql/pull/419))
+* Change ODBC version to 1.4 for release ([#542](https://github.com/opensearch-project/sql/pull/542))
```
Comment (acarbonetto) [inline release-notes/sql-odbc.OpenSearch.release-notes-1.5.0.0.md:26]: did we not create release notes for each of these...?

**Review by acarbonetto — APPROVED** [review]

File: release-notes/sql-odbc.OpenSearch.release-notes-1.5.0.0.md (hunk 3 above)
Comment (Yury-Fridlyand) [reply release-notes/sql-odbc.OpenSearch.release-notes-1.5.0.0.md:26]: No, it wasn't released. It was repo-wide version bump on SQL repo which contained ODBC as well.

**Pull request merged**